core/tests/puppeteer-acceptance-tests/user-utilities/exploration-creator-utils.ts

```diff
@@ -13,146 +13,623 @@
+    });
+
+    if (selectedLanguage.includes(expectedLanguage)) {
+      showMessage(
+        `The language ${selectedLanguage} contains the expected language.`
+      );
+    } else {
+      throw new Error('Language is not correct.');
+    }
+    await this.page.keyboard.press('Enter');
+  }
+
+  async addTags(tagNames: string[]): Promise<void> {
+    for (let i = 0; i < tagNames.length; i++) {
+      await this.clickOn(addTagsInputBox);
+      await this.type(addTagsInputBox, tagNames[i].toLowerCase());
+      await this.page.keyboard.press('Tab');
+    }
+  }
+
+  async expectTagsToMatch(expectedTags: string[]): Promise<void> {
+    /**
+     * Converting tags to lowercase because while accessing added tags with
+     * `document.querySelectorAll('mat-chip')` we get tags converted in lowercase.
+     */
+    const lowercaseExpectedTags = expectedTags.map(tag => tag.toLowerCase());
+    await this.page.waitForSelector('mat-chip-list');
+    const addedTags = await this.page.evaluate(() => {
+      const tagElements = Array.from(document.querySelectorAll('mat-chip'));
+      return tagElements
+        .map(tag => {
+          const textContent =
+            tag.querySelector('.mat-chip-remove')?.previousSibling?.textContent;
+          return textContent ? textContent.trim() : '';
+        })
+        .filter(Boolean);
     });
 
+    for (const expectedTag of lowercaseExpectedTags) {
+      if (!addedTags.includes(expectedTag)) {
+        throw new Error(`Tag "${expectedTag}" was not added.`);
+      }
+    }
+
+    showMessage('All expected tags were added successfully.');
+  }
+
+  /**
+   * This function allows you to preview the summary of exploration.
+   */
+  async previewSummary(): Promise<void> {
+    await this.clickOn(previewSummaryButton);
+    await this.expectPreviewSummaryToBeVisible();
+    await this.clickOn(dismissPreviewButton);
+  }
+
+  /**
+   * This function verifies that the preview summary is visible.
+   */
+  async expectPreviewSummaryToBeVisible(): Promise<void> {
+    await this.page.waitForSelector(dismissPreviewButton);
+    const previewSummary = await this.page.$(dismissPreviewButton);
+
+    if (previewSummary) {
+      showMessage('Preview summary is visible.');
+    } else {
+      throw new Error('Preview summary is not visible.');
+    }
+  }
+
+  /**
+   * This function enables Automatic Text-to-Speech
+   * switch present in settings tab.
+   */
+  async enableAutomaticTextToSpeech(): Promise<void> {
+    await this.clickOn(textToSpeechToggle);
+    await this.expectAutomaticTextToSpeechToBeEnable();
+  }
+
+  /**
+   * This function checks whether the Automatic Text-to-Speech
+   * setting is enabled or disabled.
+   */
+  async expectAutomaticTextToSpeechToBeEnable(): Promise<void> {
+    await this.page.waitForSelector('#text-speech-switch');
+    const autoTTSswitch = await this.page.$('#text-speech-switch');
+    const isAutoTTSwitchOn = await this.page.evaluate(
```

Comment:
autoTtsSwitchIsOn

(Let's also use Tts here and above rather than TTS -- I think the all-caps there is making it easy to omit the final S of TTS)

Reply:
done